src/cennznut/v0/contract.rs

```diff
@@ -46,7 +46,10 @@ impl Encode for Contract {
         };
         buf.push_byte(has_cooldown_byte.swap_bits());
         let address: ContractAddress = self.address;
-        buf.write(&address);
+
+        for byte in &address {
```

Comment:
[optional] I would suggest introducing a function/closure for doing this and calling it here and in the other place later down that you needed it again.